### PR TITLE
Rm family

### PR DESCRIPTION
**Pull request by rdaly525**
This PR removes 'family' user code.

-Removes family_closure from examples/alu and examples/*ir
-Changes IR constructor to pass in Peak class instead of family closure
-Changes Mapper API to use Peak class instead of family closure
-PeakMeta.rebind() is more optimized
-I cache calls to rebind so that I do not need to rerun inspect.getsource and exec
-I fixed the exec call so that error reporting actually reports the correct file and correct line
-Added api (PeakMeta.get_inputs(), and PeakMeta.get_outputs) to return the product type of the __call__ inputs and outputs respectively

### CODE REVIEW

**Comment by rdaly525** [comment]
This will be fixed once https://github.com/leonardt/hwtypes/pull/77 is in.
Unfortunately looks like this might require some magma rewrites.

@cdonovick, I would advocate creating a separate hwtypes branch so that we can make progress on auto-mapping.

File: peak/mapper/binding.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+from hwtypes import is_adt_type
+from hwtypes.adt import Product, Enum, Sum
+from .util import SubTypeDict
+
+
+__ALL__ = ['Binder', 'get_from_path', 'set_from_path', 'binding_pretty_print']
+
+def copy_smt_value(val):
+    if isinstance(val,(SMTBit,SMTBitVector)):
+        return type(val)(val.value)
```
Comment (cdonovick) [inline peak/mapper/binding.py]: This doesn't do anything meaningful unless you are going to be setting attributes on val.

File: peak/irs/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+from .coreir.ir import gen_CoreIR
```
Comment (cdonovick) [inline peak/irs/__init__.py]: should be in examples.

**Comment by cdonovick** [comment]
Also CI needs to pass

File: peak/mapper/binding.py (hunk 1 above)
Comment (rdaly525) [reply peak/mapper/binding.py]: Fixed by #76

File: peak/irs/__init__.py (hunk 2 above)
Comment (rdaly525) [reply peak/irs/__init__.py]: Fixed in last commit

**Comment by rdaly525** [comment]
I am going to move examples in a separate PR.

**Pull request merged**